aiogram/dispatcher/event/handler.py

```diff
@@ -1,45 +1,93 @@
 import asyncio
 import contextvars
 import inspect
+from contextlib import AsyncExitStack
 from dataclasses import dataclass, field
 from functools import partial
-from typing import Any, Awaitable, Callable, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, Awaitable, Callable, Dict, List, Optional, Tuple, Type, Union, cast
 
 from aiogram.dispatcher.filters.base import BaseFilter
 from aiogram.dispatcher.handler.base import BaseHandler
+from aiogram.dispatcher.requirement import (
+    CacheKeyType,
+    Requirement,
+    get_reqs_from_callable,
+    get_reqs_from_class,
+)
 
 CallbackType = Callable[..., Awaitable[Any]]
 SyncFilter = Callable[..., Any]
 AsyncFilter = Callable[..., Awaitable[Any]]
 FilterType = Union[SyncFilter, AsyncFilter, BaseFilter]
 HandlerType = Union[FilterType, Type[BaseHandler]]
 
+REQUIREMENT_CACHE_KEY = "_req_cache"
+ASYNC_STACK_KEY = "_stack"
+
+
+def _is_class_handler(handler: HandlerType) -> bool:
+    return isinstance(handler, type) and issubclass(handler, BaseHandler)
+
 
 @dataclass
 class CallableMixin:
     callback: HandlerType
     awaitable: bool = field(init=False)
     spec: inspect.FullArgSpec = field(init=False)
 
+    __reqs__: Dict[str, Requirement[Any]] = field(init=False)
```

Comment:
Why this attribute named like magic?

Reply:
Honestly, I don't recall why if I named it as a dunder attribute. Noted.